crates/ev-cli/src/commands/interact.rs

```diff
@@ -0,0 +1,201 @@
+use crate::theme::CliTheme;
+use dialoguer::{Input, Select};
+use indicatif::{ProgressBar, ProgressStyle};
+
+pub mod validators {
+    use lazy_static;
+    use regex::Regex;
+    use std::fmt::Debug;
+    use thiserror::Error;
+
+    #[derive(Debug, Error)]
+    pub enum ValidationError {
+        #[error("Value must be a valid HTTPS domain")]
+        InvalidHostname,
+        #[error("Invalid custom domain. You should not use a top-level domain name, or include a protocol or path")]
+        InvalidCustomDomain,
+        #[error("Destination Domain must be a valid hostname and not contain a protocol or path.")]
+        InvalidDestinationDomain,
+        #[error("Invalid function name. Must be between 2 and 40 characters, and contain only alphanumeric characters, dashes, and underscores")]
+        InvalidFunctionName,
+        #[error("Invalid function language. Must be one of: node@18, node@20, python@3.9, python@3.10, python@3.11")]
```

Comment:
todo: will add a section on this to docs and add the link here (also feels like this comment could drift easily, will make a note)